Cargo.toml

```diff
@@ -17,7 +17,7 @@ categories = ["embedded", "api-bindings"]
 default = ["log"]
 
 [dependencies]
-esp-idf-sys = { version = "0.33.0" }
+esp-idf-sys = { version = "0" }
```

Comment:
You have to specify a version to rely on. Otherwise `esp-idf-sys` can make breaking changes and cargo would gladly pull them in and fail to compile this library.

Cargo treats semver a bit special. The first non-zero version number is the one denoting API breakage. So even if `0.34.0` is compatible with this library, `0.35.0` might not be. So we can't "lie" and say this crate *will* be compatible with `0.x` for *any x*, because it most likely won't be, they have no reason to keep the API stable between breaking changes. For the same reason you specify that you support version `1.x` on a crate that has reached "stable 1.0", but you don't promise your crate will support `2.0`.

I'm fine with merging this if bumped to `0.34` if tested and works :)

Reply:
> This compatibility convention is different from SemVer in the way it treats versions before 1.0.0. While SemVer says there is no compatibility before 1.0.0, Cargo considers 0.x.y to be compatible with 0.x.z, where y ≥ z and x > 0.

-- https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Reply:
Hi, I've tested on xtensa and risc, so happy with this. I was fighting quite a bit with the semver on this one. I considered using `>=0.33.0, <=0.34.0` as the version string, but didn't want to run into this guideline:

> Avoid constraining the upper bound of a version to be anything less than the next semver incompatible version (e.g. avoid ">=2.0, <2.4") as other packages in the dependency tree may require a newer version, leading to an unresolvable error (see [#9029](https://github.com/rust-lang/cargo/issues/9029)). Consider whether controlling the version in your [Cargo.lock](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html) would be more appropriate.

I will revise this at your discretion. Thank you.

Reply:
PS. Great crate. Used it today, loading a binary from directly included bytes as in the example. I wanted to look into the rollback feature next.

Reply:
> I considered using >=0.33.0, <=0.34.0 as the version string

That's tricky. Just don't care about old versions and the problem goes away. Users who for some weird reason want to stay on `0.33.0` can use the current version of `esp-ota` and the rest can just move on to the latest version.

Reply:
> PS. Great crate. Used it today, loading a binary from directly included bytes as in the example. I wanted to look into the rollback feature next.

Thanks! Glad it is useful to you! After writing it I realized `esp-idf-svc` actually has OTA-bindings. But I still also use my own crate, as I find the API of this crate to be much more ergonomic than the one in `esp-idf-svc` (last time I checked)

Reply:
Done. Also find it much cleaner. And better documented. I will open another PR with an example added based on my MVP.